arangod/Auth/UserManager.cpp

```diff
@@ -58,19 +58,16 @@ namespace {
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief return a pointer to the system database or nullptr on error
 ////////////////////////////////////////////////////////////////////////////////
-arangodb::SystemDatabaseFeature::ptr getSystemDatabase() {
-  auto* feature =
-      arangodb::application_features::ApplicationServer::lookupFeature<arangodb::SystemDatabaseFeature>();
-
-  if (!feature) {
+arangodb::SystemDatabaseFeature::ptr getSystemDatabase(
+    arangodb::application_features::ApplicationServer& server) {
+  if (!server.hasFeature<arangodb::SystemDatabaseFeature>()) {
```

Comment:
To be clear, are you proposing adding back a method like `template<typename T> T* lookupFeature()` which returns a (possibly null) pointer, in addition to the `hasFeature`/`getFeature` pair? I'd prefer to avoid exposing null pointers if we can. My hope would be that after we destaticize everything, we can just cache a feature reference in a constructor/prepare when necessary for performance. (The main reason to do lookups everywhere right now is because a lot of them are in static methods.)